src/common/sphere_library/smutex.cpp

```diff
@@ -9,7 +9,9 @@ SimpleMutex::SimpleMutex()
 	InitializeCriticalSectionAndSpinCount(&m_criticalSection, 0x80000020);
 #else
 	pthread_mutexattr_init(&m_criticalSectionAttr);
+#ifndef __APPLE__
```

Comment:
Have you tried one of those solutions to keep the _settype call?
https://stackoverflow.com/questions/18375527/why-is-gnu-source-macro-required-for-pthread-mutexattr-settype-while-it-is-in